wgpu-hal/src/metal/command.rs

```diff
@@ -21,6 +21,26 @@ impl Default for super::CommandState {
     }
 }
 
+impl Drop for super::CommandState {
+    fn drop(&mut self) {
+        // Metal asserts when a MTLCommandEncoder is deallocated without a call
+        // to endEncoding. This isn't documented at
+        // https://developer.apple.com/documentation/metal/mtlcommandencoder?language=objc
```

Comment:
Oh, I found the rule in the docs here:
https://developer.apple.com/documentation/metal/mtlcomputecommandencoder#overview

> You must always call [endEncoding()](https://developer.apple.com/documentation/metal/mtlcommandencoder/1458038-endencoding) before the encoder is released or before creating another encoder.

Reply:
That's just for `MTLComputeCommandEncoder`. There does not seem to be a similar rule mentioned in the corresponding sections for `MTLRenderCommandEncoder` or `MTLBlitCommandEncoder`, but our crashes certainly establish that a similar rule holds for them as well.

Reply:
We still need the link I found mentioned in the comment.